src/widget/widget.js

```diff
@@ -140,4 +187,69 @@ export default class Widget extends Plugin {
 	_setSelectionOverElement( element ) {
 		this.editor.document.selection.setRanges( [ ModelRange.createOn( element ) ] );
 	}
+
+	/**
+	 * Checks if {@link module:engine/model/element~Element element} placed next to the current
+	 * {@link module:engine/model/selection~Selection model selection} exists and is marked in
+	 * {@link module:engine/model/schema~Schema schema} as `object`.
+	 *
+	 * @private
+	 * @param {Boolean} forward Direction of checking.
+	 * @returns {module:engine/model/element~Element|null}
+	 */
+	_getObjectElementNextToSelection( forward ) {
+		const modelDocument = this.editor.document;
+		const schema = modelDocument.schema;
+		const modelSelection = modelDocument.selection;
+		const dataController = this.editor.data;
+
+		// Clone current selection to use it as a probe. We must leave default selection as it is so it can return
+		// to its current state after undo.
+		const probe = ModelSelection.createFromSelection( modelSelection );
+		dataController.modifySelection( probe, { direction: forward ? 'forward' : 'backward' } );
+		const objectElement = forward ? probe.focus.nodeBefore : probe.focus.nodeAfter;
+
+		if ( objectElement instanceof ModelElement && schema.objects.has( objectElement.name ) ) {
+			return objectElement;
+		}
+
+		return null;
+	}
+}
+
+// Returns the selected element. {@link module:engine/model/element~Element Element} is considered as selected if there is only
+// one range in the selection, and that range contains exactly one element.
+// Returns `null` if there is no selected element.
+//
+// @param {module:engine/model/selection~Selection} modelSelection
+// @returns {module:engine/model/element~Element|null}
+function getSelectedElement( modelSelection ) {
```

Comment:
I was wondering whether this should stay here or be moved as model's selection method. We have similar one in view's selection.